Source/AtomicApp/IPCServerApp.cpp

```diff
@@ -0,0 +1,261 @@
+
+#include <Atomic/Input/InputEvents.h>
+
+#include "IPCServerApp.h"
+
+#pragma push_macro("PostMessage")
+#undef PostMessage
+
+using namespace ToolCore;
+
+namespace Atomic
+{
+
+    IPCServerApp::IPCServerApp(Context* context) :
+        AppBase(context),
+        playerEnabled_(false)
+    {
+    }
+
+    IPCServerApp::~IPCServerApp()
+    {
+    }
+
+    void IPCServerApp::Setup()
+    {
+        AppBase::Setup();
+
+        // Register IPC system
+        context_->RegisterSubsystem(new IPC(context_));
+
+        ToolEnvironment* env = new ToolEnvironment(context_);
+        context_->RegisterSubsystem(env);
+
+        env->Initialize();
+
+        ToolSystem* system = new ToolSystem(context_);
+        context_->RegisterSubsystem(system);
+
+        engineParameters_["Headless"] = true;
+        engineParameters_["LogLevel"] = LOG_INFO;
+    }
+
+    void IPCServerApp::Stop()
+    {
+        IPC* ipc = GetSubsystem<IPC>();
+
+        if (ipc)
+        {
+            ipc->Shutdown();
+        }
+
+        AppBase::Stop();
+    }
+
+    bool IPCServerApp::RunIPCPlayer(const String& projectName, const String& projectPath, const String &addArgs)
+    {
+        /* Params to pass in (possibly) from PE
+        --windowposx 244
+        --windowposy 157
+        --windowwidth 1440
+        --windowheight 806
+        --resizable
+
+        ALSO STUFF TO TELL VSEATOMIC WHICH CUTSCENE TO LAUNCH..app delegate will need to parse run params I think
+
+        */
+
+        if (playerBroker_.NotNull())
+            return false;
+
+        FileSystem* fileSystem = GetSubsystem<FileSystem>();
+        ToolSystem* tsystem = GetSubsystem<ToolSystem>();
+        ToolEnvironment* tenv = GetSubsystem<ToolEnvironment>();
+
+        String projectAssembly = projectName + ".dll";
+        String projectExe = projectName + ".exe";
+
+        String playerBinary = "";
+
+        String resourcePath = projectPath + "Resources/" + projectAssembly;
+
+        // TODO: We need to configure project as managed
+        if (fileSystem->FileExists(resourcePath))
+        {
+#ifdef ATOMIC_DEV_BUILD
+
+#ifdef ATOMIC_DEBUG        
+            playerBinary = projectPath + "AtomicNET/Debug/Bin/Desktop/" + projectExe;
+#else
+            playerBinary = projectPath + "AtomicNET/Release/Bin/Desktop/" + projectExe;
+#endif
+
+#else
+            // TODO: We are using the release build of the managed project here, how and when to use debug?
+            playerBinary = projectPath + "AtomicNET/Release/Bin/Desktop/" + projectExe;
+#endif
+
+
+            if (!fileSystem->FileExists(playerBinary))
+            {
+                ATOMIC_LOGERRORF("Managed player: %s does not exist", playerBinary.CString());
+                return false;
+            }
+
+        }
+        else
+        {
+            return false;
+        }
+
+        Vector<String> vargs;
+
+        String args = ToString("--player --project \"%s\"", projectPath.CString());
```

Comment:
I think the --project path is valid here, I wonder about the --player arg, is it necessary, if so maybe a rename.  It is necessary with the AtomicEditor as that is a single binary which can run in editor/player mode, for example.